pvlib/pvsystem.py

```diff
@@ -1967,174 +2016,18 @@ def sapm(effective_irradiance, temp_cell, module):
-
-    Parameters
-    ----------
-    poa_global : numeric
-        Total incident irradiance in W/m^2.
-
-    temp_air : numeric
-        Ambient dry bulb temperature in degrees C.
-
-    wind_speed : numeric, default 1.0
-        Wind speed in m/s measured at the same height for which the wind loss
-        factor was determined.  The default value is 1.0, which is the wind
-        speed at module height used to determine NOCT.
-
-    eta_m : numeric, default 0.1
-        Module external efficiency as a fraction, i.e., DC power / poa_global.
-
-    alpha_absorption : numeric, default 0.9
-        Absorption coefficient
-
-    model_params : string, tuple, or list (no dict), default 'freestanding'
-        Heat loss factors to be used.
-
-        If string, can be:
-
-            * 'freestanding' (default)
-                Modules with rear surfaces exposed to open air (e.g. rack
-                mounted).
-            * 'insulated'
-                Modules with rear surfaces in close proximity to another
-                surface (e.g. roof mounted).
-
-        If tuple/list, supply parameters in the following order:
-
-            * constant_loss_factor : float
-                Combined heat loss factor coefficient. Freestanding
-                default is 29, fully insulated arrays is 15.
-
-            * wind_loss_factor : float
-                Combined heat loss factor influenced by wind. Default is 0.
-
-    Returns
-    -------
-    temp_cell : numeric or Series
-        Cell temperature in degrees Celsius
-
-    References
-    ----------
-    [1]"PVsyst 6 Help", Files.pvsyst.com, 2018. [Online]. Available:
-    http://files.pvsyst.com/help/index.html. [Accessed: 10- Dec- 2018].
-
-    [2] Faiman, D. (2008). "Assessing the outdoor operating temperature of
-    photovoltaic modules." Progress in Photovoltaics 16(4): 307-315.
-    """
-
-    pvsyst_presets = TEMP_MODEL_PARAMS['pvsyst']
-
-    if isinstance(model_params, str):
-        model_params = model_params.lower()
-        constant_loss_factor, wind_loss_factor = pvsyst_presets[model_params]
-    elif isinstance(model_params, (tuple, list)):
-        constant_loss_factor, wind_loss_factor = model_params
-    else:
-        raise TypeError(
-            "Please provide model_params as a str, or tuple/list."
-        )
+sapm_celltemp = deprecated('0.7', alternative='temperature.sapm_cell',
+                           name='sapm_celltemp', removal='0.8',
+                           addendum='Note that the argument names and order '
+                           'for temperature.sapm_cell are different than for '
+                           'sapm_celltemp')(temperature.sapm_cell)
 
-    total_loss_factor = wind_loss_factor * wind_speed + constant_loss_factor
-    heat_input = poa_global * alpha_absorption * (1 - eta_m)
-    temp_difference = heat_input / total_loss_factor
-    temp_cell = temp_air + temp_difference
 
-    return temp_cell
+pvsyst_celltemp = deprecated(
```

Comment:
are these deprecations tested with functions decorated with `fail_on_pvlib_version(0.8)`? this ensures that we eventually remove the deprecated functions.

Reply:
of course I find the function in test_temperature.py right after this...

Does make me wonder... is there any point in keeping the old functions around if the argument order has changed? Probably better for a user to see an exception than wrong results and a warning.

Reply:
I think the functions are removed, the `PVsystem` method is still `sapm_celltemp`.

Reply:
The `deprecated` function returns a function, so this line defines the function `pvlib.pvsystem.sapm_celltemp`. The effect is something like this:

```python
def sapm_celltemp(*args, **kwargs):
    warings.DeprecationWarning('the message')
    return temperature.sapm_celltemp(*args, **kwargs)
```

I'm suggesting that this behavior is worse than breaking user code without a warning. 

I think it could be possible to define a translation function between to two signatures. Something like

```python
def _sapm_celltemp_translator(*args, **kwargs):
    reordered_args = 
    renamed_kwargs = 
    return temperature.sapm_celltemp(*reordered_args, **renamed_kwargs)

sapm_celltemp = deprecated('0.7', alternative='temperature.sapm_cell',
                           name='sapm_celltemp', removal='0.8',
                           addendum='Note that the argument names and order '
                           'for temperature.sapm_cell are different than for '
                           'sapm_celltemp')(_sapm_celltemp_translator)
```